### DISCUSSION TITLE
Event that fires on every controller call?

### DISCUSSION THREAD

**Discussion by hallleron** [pimcore/pimcore#10085] [Q&A / Support]
Hey developers,

I have a question regarding events. I need an event that fires on every controller call in the frontend view. I looked through the official event list, but didn't find one that suit my needs ([https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html)). Has anybody got an idea?

Thanks in advance!

**Comment by SQRR** [comment]
Jep, I'm looking for the same.
It used to be "onKernelController" but that doesn't trigger anymore.

**Reply by fashxp** [reply]
controller needs to implement the `KernelControllerEventInterface` interface.

**Reply by hallleron** [reply]
> controller needs to implement the `KernelControllerEventInterface` interface.

I don't really get it. I have an event listener and want to detect every controller request in the frontend. So I don't have a controller in my bundle, only an event listener.

**Reply by dpfaffenbauer** [reply]
https://symfony.com/doc/current/reference/events.html#kernel-controller

**Reply by fashxp** [reply]
sorry, thought you were using the `onKernelController` method on the controller. for event listeners @dpfaffenbauer gave the answer.

**Reply by hallleron** [reply]
@dpfaffenbauer Danke, das wars! =)
